
<launch>
  <arg name="base_topic" default="stereo_down" />
  <arg name="camera" default="BB2_08S2C_25_1" />
  <arg name="calibration" default="boxed_water" />
  <arg name="camera_params" default="$(find xiroi)/config/cameras/config/stereo_down_params.yaml" />
  <arg name="processing" default="false"/>
  <arg name="decimate" default="false"/>

  <node pkg="camera1394stereo" type="camera1394stereo_node" name="stereo_down_node" output="screen" >
      <rosparam file="$(arg camera_params)" />
      <param name="frame_id" value="$(arg base_topic)/left_optical" />
      <param name="camera_info_url_left" value="file://$(find xiroi)/config/cameras/$(arg camera)/calibration_$(arg calibration)_left.yaml" />
      <param name="camera_info_url_right" value="file://$(find xiroi)/config/cameras/$(arg camera)/calibration_$(arg calibration)_right.yaml" />
      <remap from="stereo_camera" to="$(arg base_topic)" />
  </node>
  
  <group ns="stereo_down" if="$(arg processing)">
    <node name="stereo_image_proc" pkg="stereo_image_proc" type="stereo_image_proc">
      <rosparam file="$(find xiroi)/config/cameras/config/disparity_params.yaml"/>
    </node>
  </group>

  <include if="$(arg decimate)" file="$(find xiroi)/launch/modules/cameras/stereo_crop_decimate.launch" >
      <arg name="stereo" value="stereo_down" />
      <arg name="decimation" value="2" />
    </include>

  <group ns="stereo_down/scaled_x2" if="$(arg processing)">
    <node name="stereo_image_proc" pkg="stereo_image_proc" type="stereo_image_proc">
      <rosparam file="$(find xiroi)/config/cameras/config/disparity_params_x2.yaml"/>
    </node>
  </group>

  
  <!--<node pkg="xiroi" type="image_throttle.py" name="image_throttle" output="screen" >
  </node>-->

</launch>